artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagedMessageImpl.java

```diff
@@ -59,8 +61,19 @@ public PagedMessageImpl(final Message message, final long[] queueIDs) {
       this.message = message;
    }
 
-   public PagedMessageImpl(StorageManager storageManager) {
+   public PagedMessageImpl(int storedSize, StorageManager storageManager) {
       this.storageManager = storageManager;
+      this.storedSize = storedSize;
+   }
+
+
+   @Override
+   public int getStoredSize() {
+      if (storedSize <= 0) {
```

Comment:
Instead of using this logic, woudn't simpler to correctly initialize `storedSize` for both protocols: for core `== encodeSize`, for AMQP, using a different value?

Reply:
I am not sure it's safe to do that in case the encodeSize changed between the contruction and when getStoredSize is used.

We can change this If you're sure it's safe. I wasn't bold enough.

Reply:
> We can change this If you're sure it's safe. I wasn't bold enough.

Agree, I'm not sure of it as well (let me check around on the code)